org.envirocar.app/src/org/envirocar/app/views/carselection/CarSelectionActivity.java

```diff
@@ -300,10 +300,19 @@ public void onCarAdded(Car car) {
             mCarListAdapter.addCarItem(car);
             showSnackbar(String.format(getString(R.string.car_selection_successfully_added_tmp),
                     car.getManufacturer(), car.getModel()));
+
+            // Check the total Cars count after adding, if only 1 car then set it.
+            int count = mCarListAdapter.getCount();
+            if(count == 1) {
+                mCarManager.setCar(car);
+            }
         } else {
             showSnackbar(String.format(getString(R.string.car_selection_already_in_list_tmp),
                     car.getManufacturer(), car.getModel()));
         }
+        finish();
+        overridePendingTransition(0, 0);
+        startActivity(getIntent());
```

Comment:
Restarting the activity to refresh the view is not a good practice. I would rather suggest you to find a way to set the corresponding RadioButton as selected for the selected car.